subsys/bootloader/cmake/sign.cmake

```diff
@@ -67,32 +81,71 @@ endif()
 
 add_custom_command(
   OUTPUT
-  ${signature_file}
+  ${HASH_FILE}
   COMMAND
-  ${hashcmd}
+  ${HASH_CMD}
+  DEPENDS
+  ${KERNEL_ELF_TO_BE_SIGNED}
+  WORKING_DIRECTORY
+  ${PROJECT_BINARY_DIR}
+  COMMENT
+  "Creating hash of application"
+  USES_TERMINAL
+  )
+
+add_custom_target(
+  hash_file_target
+  DEPENDS
+  ${HASH_FILE}
+  ${KERNEL_ELF_TO_BE_SIGNED}
+  generated_file_directory_target
+  )
+
+add_custom_command(
+  OUTPUT
+  ${SIGNATURE_FILE}
   COMMAND
-  ${signcmd}
+  ${SIGN_CMD}
   DEPENDS
-  ${sign_depends}
-  WORKING_DIRECTORY ${PROJECT_BINARY_DIR}
+  ${KERNEL_ELF_TO_BE_SIGNED}
+  ${HASH_FILE}
+  WORKING_DIRECTORY
+  ${PROJECT_BINARY_DIR}
   COMMENT
   "Creating signature of application"
   USES_TERMINAL
   )
 
+
+add_custom_target(
+  signature_file_target
+  DEPENDS
+  ${SIGN_KEY_FILE_DEPENDS}
+  ${SIGNATURE_FILE}
+  hash_file_target
+  )
+
 if (CONFIG_SB_PRIVATE_KEY_PROVIDED)
   add_custom_command(
     OUTPUT
     ${SIGNATURE_PUBLIC_KEY_FILE}
     COMMAND
-    ${pubgencmd}
+    ${PUB_GEN_CMD}
     DEPENDS
-    ${sign_depends}
-    WORKING_DIRECTORY ${PROJECT_BINARY_DIR}
+    ${SIGNATURE_FILE}
```

Comment:
You should depend on `signature_file_target` instead of the file directly. This because dependencies to targets are handled better in cmake than dependencies to files.

Reply:
fixed this